modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/validators.js

```diff
@@ -0,0 +1,366 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const ERROR_OBJECT_OF_TYPE = 'Expected object of one type.';
+const ERROR_ONE_OF = 'Expected one of the following values:';
+const ERROR_ONE_OF_TYPE = 'Expected one of given types.';
+
+function _getFunctionName(fn) {
+	if (!fn.name) {
+		const stringifiedName = fn.toString();
+
+		fn.name = stringifiedName.substring(9, stringifiedName.indexOf('('));
+	}
+
+	return fn.name;
+}
+
+function _isDefAndNotNull(value) {
+	return value !== undefined && value !== null;
+}
+
+/**
+ * Provides access to various type validators that will return an
+ * instance of Error when validation fails. Note that all type validators
+ * will also accept null or undefined values. To not accept these you should
+ * instead make your state property required.
+ */
+const validators = {
+	any: () => () => true,
+	array: buildTypeValidator('array'),
+
+	/**
+	 * Creates a validator that checks that the value it receives is an array
+	 * of items, and that all of the items pass the given validator.
+	 * @param {!Object} validator Validator to check each item against.
+	 * @return {!function()}
+	 */
+	arrayOf(validator) {
+		if (isInvalid(validators.object(validator))) {
+			throwConfigError('object', validator, 'arrayOf');
+		}
+
+		return maybe((value, name, context) => {
+			const result = validators.array(value, name, context);
+			if (isInvalid(result)) {
+				return result;
+			}
+
+			return validateArrayItems(validator, value, name, context);
+		});
+	},
+	bool: buildTypeValidator('boolean'),
+	func: buildTypeValidator('function'),
+
+	/**
+	 * Creates a validator that checks for a value within a range.
+	 * @param {!Number} min The minimum value allowed.
+	 * @param {!Number} max The maximum value allowed.
+	 * @return {!function()}
+	 */
+	inRange(min, max) {
```

Comment:
We should [avoid abbreviations](https://github.com/liferay/liferay-frontend-projects/blob/master/guidelines/general/naming.md).

So `min` → `minimum` etc.

Exceptions would be things like `fn` earlier in this file (because you can't write `function` as it is a keyword).